File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/old/wheelTest.java
package org.firstinspires.ftc.teamcode;

import com.qualcomm.robotcore.eventloop.opmode.Disabled;
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
import com.qualcomm.robotcore.hardware.DcMotor;
import com.qualcomm.robotcore.hardware.DcMotorSimple;

@TeleOp(name="wheel test", group="Linear OpMode")
public class wheelTest extends LinearOpMode {

    private double speed = 0.2;
    class Motor {
        DcMotor drive;
        Motor(String name, DcMotorSimple.Direction direction) {
            this.drive = hardwareMap.get(DcMotor.class, name);
            this.drive.setDirection(direction);
            this.drive.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
            this.drive.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
        }
    }
    @Override
    public void runOpMode() {
        Motor elevator = new Motor("elevator", DcMotorSimple.Direction.REVERSE);
        elevator.drive.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.FLOAT);
        waitForStart();
        while (opModeIsActive()) {
            telemetry.addLine(String.valueOf(elevator.drive.getCurrentPosition()));
            telemetry.update();
        }
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/oldtests/WheelTest.kt
package org.firstinspires.ftc.teamcode.oldtests

import com.qualcomm.robotcore.eventloop.opmode.Disabled
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode
import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import com.qualcomm.robotcore.hardware.DcMotorSimple
import org.firstinspires.ftc.teamcode.Wheel

@TeleOp(name = "wheel test", group = "Linear OpMode")
@Disabled
class WheelTest : LinearOpMode() {
    override fun runOpMode() {
        val wheels = arrayOf(
            Wheel("m1", hardwareMap, DcMotorSimple.Direction.REVERSE),
            Wheel("m2", hardwareMap, DcMotorSimple.Direction.FORWARD),
        )
        waitForStart()
        while (opModeIsActive()) {
            wheels[0].setPower(gamepad1.left_trigger * 0.1)
            wheels[1].setPower(gamepad1.right_trigger * 0.1)
            for (i in wheels) {
                telemetry.addData("position",i.getPosition())
            }
            telemetry.update()
        }
    }
}